### PR TITLE
Fully map n/m/{village, stat, advancement, inventory}

### PR DESCRIPTION
**Pull request by pyrox0**
Four easy checkboxes in the classes panel, I'm happy with that.

### CODE REVIEW

**Comment by eristhea** [comment]
I knew the checkboxes were a brilliant idea

**Comment by pyrox0** [comment]
Updated to 24w05a and all invalid mappings/merge conflicts fixed :tada:

File: mappings/net/minecraft/village/TradeOffers.mapping [hunk 1]
```diff
@@ -90,8 +92,22 @@ CLASS net/minecraft/unmapped/C_hqrlptsj net/minecraft/village/TradeOffers
 		METHOD <init> (Lnet/minecraft/unmapped/C_gmbqjnle;IILnet/minecraft/unmapped/C_sddaxwyk;IIIF)V
 			ARG 1 item
 			ARG 2 count
+			ARG 3 price
 			ARG 4 processedItem
 			ARG 5 processedItemCount
+			ARG 6 maxUses
+			ARG 7 experience
+			ARG 8 multiplier
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/village/TradeOffers.mapping]: Multiplier for what? Is there a more descriptive name that makes sense?
(On mobile, can't check source)

File: mappings/net/minecraft/village/TradeOffers.mapping [hunk 2]
```diff
@@ -90,8 +92,22 @@ CLASS net/minecraft/unmapped/C_hqrlptsj net/minecraft/village/TradeOffers
...
+		METHOD <init> (Lnet/minecraft/unmapped/C_gmbqjnle;IILnet/minecraft/unmapped/C_sddaxwyk;IIIFZ)V
+			ARG 1 item
+			ARG 2 buyCount
+			ARG 3 price
+			ARG 4 stack
+			ARG 5 sellCount
+			ARG 6 maxUses
+			ARG 7 experience
+			ARG 8 multiplier
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/village/TradeOffers.mapping]: Same question

File: mappings/net/minecraft/village/TradeOffers.mapping [hunk 3]
```diff
@@ -101,6 +117,16 @@ CLASS net/minecraft/unmapped/C_hqrlptsj net/minecraft/village/TradeOffers
 			ARG 6 maxUses
 			ARG 7 experience
 			ARG 8 multiplier
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/village/TradeOffers.mapping]: Same

File: mappings/net/minecraft/village/TradeOffers.mapping [hunk 4]
```diff
@@ -101,6 +117,16 @@ CLASS net/minecraft/unmapped/C_hqrlptsj net/minecraft/village/TradeOffers
...
+		METHOD <init> (Lnet/minecraft/unmapped/C_gmbqjnle;IILnet/minecraft/unmapped/C_vorddnax;IIIFZ)V
+			ARG 1 itemConvertible
+			ARG 2 buyCount
+			ARG 3 price
+			ARG 4 item
+			ARG 5 sellCount
+			ARG 6 maxUses
+			ARG 7 experience
+			ARG 8 multiplier
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/village/TradeOffers.mapping]: Same

**Review by supersaiyansubtlety — COMMENTED** [review]
Just one question*4

File: mappings/net/minecraft/village/TradeOffers.mapping (hunk 3 above)
Comment (eristhea) [reply mappings/net/minecraft/village/TradeOffers.mapping]: I mapped this, I'm pretty sure it's for experience (also cannot check source rn)

**Review by eristhea — APPROVED** [review]
looks good, the multiplier field should definitely be clarified!

File: mappings/net/minecraft/village/TradeOffers.mapping (hunk 3 above)
Comment (pyrox0) [reply mappings/net/minecraft/village/TradeOffers.mapping]: checked source, it's for the price. Updated the name everwhere in the file!

**Comment by pyrox0** [comment]
All fields named `multiplier` are for price multipliers(found from the field `net/minecraft/village/TradeOffer.f_izfgozck`), updated everywhere in the file to reflect that!

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/village/TradeOffer.mapping [hunk 5]
```diff
@@ -64,6 +69,9 @@ CLASS net/minecraft/unmapped/C_argwryfu net/minecraft/village/TradeOffer
 	METHOD m_jrpzwrcp depleteBuyItems (Lnet/minecraft/unmapped/C_sddaxwyk;Lnet/minecraft/unmapped/C_sddaxwyk;)Z
 		ARG 1 firstBuyStack
 		ARG 2 secondBuyStack
+	METHOD m_jwtgudye read (Lnet/minecraft/unmapped/C_bngyzsts;)Lnet/minecraft/unmapped/C_argwryfu;
+		ARG 0 buf
+	METHOD m_jxiehhbs getIgnoreTags ()Z
```
Comment (IotaBread) [inline mappings/net/minecraft/village/TradeOffer.mapping]:
```suggestion
	METHOD m_jxiehhbs shouldIgnoreTags ()Z
```

Comment (Platymemo) [reply mappings/net/minecraft/village/TradeOffer.mapping]: If we're moving away from `get_` terminology, maybe `ignoresTags` or even `ignoresNbt` would be more concise.

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/village/TradeOffer.mapping]: AFAIK we're not, it's just slightly different for boolean getters (is/does/are)

File: mappings/net/minecraft/inventory/JukeboxInventory.mapping [hunk 6]
```diff
@@ -1 +1,8 @@
 CLASS net/minecraft/unmapped/C_rsrfiyeu net/minecraft/inventory/JukeboxInventory
+	METHOD m_bkmnewbu setStack (Lnet/minecraft/unmapped/C_sddaxwyk;)V
+	METHOD m_brueltfr emptyStack ()Lnet/minecraft/unmapped/C_sddaxwyk;
+	METHOD m_htpnuxlo decreaseStack (I)Lnet/minecraft/unmapped/C_sddaxwyk;
+		ARG 1 count
+	METHOD m_xwjsgxrp getStack ()Lnet/minecraft/unmapped/C_sddaxwyk;
+	CLASS C_ecqzlsuf
+		METHOD m_ejhbwvaa getBlock ()Lnet/minecraft/unmapped/C_kvegafmh;
```
Comment (Platymemo) [inline mappings/net/minecraft/inventory/JukeboxInventory.mapping]:
```suggestion
		METHOD m_ejhbwvaa getBlockEntity ()Lnet/minecraft/unmapped/C_kvegafmh;
```

File: mappings/net/minecraft/village/TradeOffer.mapping [hunk 7]
```diff
@@ -7,7 +7,9 @@ CLASS net/minecraft/unmapped/C_argwryfu net/minecraft/village/TradeOffer
 	FIELD f_lgvzktct demandBonus I
 	FIELD f_mgalkhcf maxUses I
 	FIELD f_orryzsqt merchantExperience I
+	FIELD f_qxgedhuk ignoreTags Z
```
Comment (Platymemo) [inline mappings/net/minecraft/village/TradeOffer.mapping]:
```suggestion
	FIELD f_qxgedhuk ignoreNbt Z
```
I know this is referred to as "ignore_tags" in strings, but we call it nbt everywhere else anyway, so I feel this is more clear

File: mappings/net/minecraft/inventory/Inventory.mapping [hunk 8]
```diff
@@ -1,8 +1,12 @@
 CLASS net/minecraft/unmapped/C_pjtstjoq net/minecraft/inventory/Inventory
 	FIELD f_afhsixex MAX_COUNT_PER_STACK I
+	FIELD f_vlssyyaw DEFAULT_MAX_INTERACTION_RANGE F
 	METHOD m_akaynoad setStack (ILnet/minecraft/unmapped/C_sddaxwyk;)V
 		ARG 1 slot
 		ARG 2 stack
+	METHOD m_crvopqlx canPlayerUse (Lnet/minecraft/unmapped/C_kvegafmh;Lnet/minecraft/unmapped/C_jzrpycqo;F)Z
+		ARG 0 entity
+		ARG 2 range
```
Comment (Platymemo) [inline mappings/net/minecraft/inventory/Inventory.mapping]: This is more like an added range, it looks to be in addition to the base player interaction range.
```suggestion
		ARG 2 additionalRange
```

**Comment by pyrox0** [comment]
All suggestions implemented and pushed ^^

**Pull request merged**